frontend/apps/reader/modules/readeruserhyph.lua

```diff
@@ -0,0 +1,372 @@
+local DataStorage = require("datastorage")
+local Event = require("ui/event")
+local FFIUtil = require("ffi/util")
+local InfoMessage = require("ui/widget/infomessage")
+local InputDialog = require("ui/widget/inputdialog")
+local MultiConfirmBox = require("ui/widget/multiconfirmbox")
+local UIManager = require("ui/uimanager")
+local T = require("ffi/util").template
+local Utf8 = require("utf8")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local TimeVal = require("ui/timeval")
+local lfs = require("libs/libkoreader-lfs")
+local logger = require("logger")
+local _ = require("gettext")
+
+local ReaderUserHyph = WidgetContainer:new{
+    KEEP_ORIGINAL = 1, -- for dictionary file operations
+    USE_NEW = 2,       --     -"-
+}
+
+-----------------------------------------------
+--
+-----------------------------------------------
+
+-- returns path to the user dictionary
+function ReaderUserHyph:getDictionaryPath()
+    return FFIUtil.joinPath(DataStorage:getSettingsDir(),
+        self.ui.document:getTextMainLangDefaultHyphDictionary():gsub(".pattern$", "") .. "-user.hyph")
+end
+
+-- Load the user dictionary suitable for the actual language
+-- Unload is done automatically when a new dictionary is loaded.
+function ReaderUserHyph:loadDictionary()
+    if G_reader_settings:isTrue("hyph_user_dict") then
+        self.ui.document:setUserHyphenationDict(self:getDictionaryPath())
```

Comment:
You let the file existence check and warning to crengine ?
Might be saner to check it here, no?